src/extension/php_objects.c

```diff
@@ -3,17 +3,30 @@
 //
 // This product includes software developed at Datadog
 // (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
-#include "php_objects.h"
+#include <zend_API.h>
+#include <zend_alloc.h>
+
 #include "attributes.h"
 #include "ddappsec.h"
 #include "dddefs.h"
 #include "php_compat.h"
+#include "php_objects.h"
 
 static int _module_number;
 static zend_llist _function_entry_arrays;
 
 static void _unregister_functions(void *zfe_arr_vp);
 
+typedef struct _dd_registered_entries {
+    char *nullable name;
+    size_t name_len;
+    char *nullable env_name;
+} dd_registered_entries;
+
+#define DD_MAX_REGISTERED_ENTRIES 160
+static uint8_t registered_entries_count = 0;
+static dd_registered_entries registered_entries[DD_MAX_REGISTERED_ENTRIES];
```

Comment:
This approach seems OK, though I personally would think it it simpler to store nothing whatsoever and just iterate through all the ini entries (the relevant ones could be filtered by comparing the `on_modify` function pointer, and the env variable name could just be recalculated). But two things:

1. The array size seems oversized. We don't have anywhere near 160 ini entries.
2. You're not doing any cleanup on shutdown, which probably means you'll be leaking memory on apache reloads.

Reply:
I have reduced the size and also clean all the registered entries on `mshutdown`